.github/workflows/coverage.yml

```diff
@@ -3,6 +3,9 @@ on:
   push:
   pull_request:
     branches: ['main']
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
```

Comment:
Just to check: `github.ref` gets the contributor's branch name `refs/heads/<branch>`, right, not the commit?

Reply:
I _think_ `github.ref` might be the target branch for the PR, whereas `github.head_ref` is the contributor's branch? I might be misreading this https://docs.github.com/en/actions/learn-github-actions/contexts#github-context.

Reply:
Oh yeah, sorry - I'd misread the docs and got them back to front (and so confused myself).

Reply:
They're honestly not super readable IMO 😅